content/md/articles/language/core_overview.md

```diff
@@ -194,48 +215,49 @@ If a third expression is provided and the first expression returns `nil` or
 ```
 
 ```klipse-clojure
-(if (< 10 9) "second" "third") ; (< 9 10) returns false
+(if (< 10 9) "second" "third") ; (< 10 9) returns false
 ```
 
 ```klipse-clojure
-(if (seq '()) "second") ; seq returns nil for an empty sequence
+; seq returns nil for an empty sequence, 
+; so the third expression (defaulted to nil) is returned
+(if (seq '()) "second")
 ```
 
 ```klipse-clojure
 (if (nil? (= 1 2)) "second" "third") ; differentiate between nil and false if needed
 ```
 
+```klipse-clojure
+; to do multiple things based on the condition, use a `do`
+(if (> 4 3)
+  (do
+    (println "4 is greater than 3...")
+    (println "so 3 is less than 4")
+    "4!")
+  (println "Uh oh, math is broken."))
+```
+
 <a id="when_desc"></a>
 ### when
 
 ```clojure
-([test & body])
+(when test body*)
 ```
 
-`when` takes one or more expressions.
-
-`when` provides an implicit `do` form that wraps the second and any subsequent
-expressions, and is evaluated if the first expression returns truthy --
-anything except `nil` or `false` -- otherwise `nil` is returned.
+`when` takes a test and zero or more `body` expressions. If the test returns a truthy value, all the `body` 
+expressions are evaluated in an implicit `do`. If the test is not truthy, `nil` is returned.
```

Comment:
Maybe add "(and none of the `body` expressions are evaluated)" for the `nil` return sentence?

Reply:
I added this, and it got me thinking "this is just a macro, so we could also add a brief explanation in terms of `if`", and show a macroexpand to show that it truly is just an if. Do you think this is too much too early?

Reply:
I think that's fine. Folks can either think "oh, that's interesting!" or ignore it 😄